cli/src/commands/project.rs

```diff
@@ -142,17 +143,21 @@ pub async fn handle_project(api: &mut PhylumApi, matches: &clap::ArgMatches) ->
             }
         };
 
-        for threshold_name in vec![
+        for threshold_name in &[
             "total project",
             "author",
             "engineering",
             "license",
             "malicious code",
             "vulnerability",
-        ]
-        .iter()
-        {
-            let (threshold, action) = prompt_threshold(threshold_name).unwrap_or((0, "none"));
+        ] {
+            let threshold = match prompt_threshold(threshold_name) {
+                Ok(threshold) => threshold,
+                Err(_) => {
+                    print_user_failure!("Failed to read user input");
+                    continue;
```

Comment:
Is it right that if an error occurs it is just reported to stderr and then we go on to the next threshold without updating `user_settings`? Would it be better if the user was prompted again or if an error was propagated so the operation aborts? It's similar to what the old code did so maybe it's okay.

Reply:
Yes, reporting to stderr and going to the next without updating settings is the goal. You wouldn't want to update them if the user failed to produce a valid input I'd imagine.

The previous version actually was a bit more devious. If it failed for some reason (generally this shouldn't fail), it would disable the threshold. I've changed that because I think it's usually not a good idea to change something when an error occurred.